nevergrad/optimization/optimizerlib.py

```diff
@@ -1815,6 +1816,154 @@ def __init__(
+        candidate._meta["x_probe"] = x_probe
+        return candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        self._newX.append(candidate._meta["x_probe"])
+        self._losses.append(loss)
+        if not self._buffer:
+            if "x_probe" in candidate._meta:
+                self._alg.tell(self._newX, self._losses)
+            else:
+                data = candidate.get_standardized_data(reference=self.parametrization)
+                # Tell not asked:
+                self._alg.tell(np.arctan(data), loss)
+
+    def _internal_tell_not_asked(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        raise errors.TellNotAskedNotSupportedError
+
+
+class ParametrizedBayesOptim(base.ConfiguredOptimizer):
+    """
+    Algorithms from bayes-optim package.
+
+    We use:
+    - BO
+    - PCA-BO: Principle Component Analysis (PCA) Bayesian Optimization for dimensionality reduction in BO
+
+    References
+
+    [RaponiWB+20]
+        Raponi, Elena, Hao Wang, Mariusz Bujny, Simonetta Boria, and Carola Doerr.
+        "High dimensional bayesian optimization assisted by principal component analysis."
+        In International Conference on Parallel Problem Solving from Nature, pp. 169-183.
+        Springer, Cham, 2020.
+
+
+    Parameters
+    ----------
+    init_budget: int or None
+        Number of initialization algorithm steps
+    pca: bool
+        whether to use the PCA transformation defining PCA-BO rather than BO
+    n_components: float or 0.95
+        Principal axes in feature space, representing the directions of maximum variance in the data.
+        It represents the percentage of explained variance
+    prop_doe_factor: float or None
+        Percentage of the initial budget used for DoE and eventually overwriting init_budget
+    """
+
+    no_parallelization = True
+    recast = True
+
+    # pylint: disable=unused-argument
+    def __init__(
+        self,
+        *,
+        init_budget: tp.Optional[int] = None,
+        pca: tp.Optional[bool] = False,
+        n_components: tp.Optional[float] = 0.95,
+        prop_doe_factor: tp.Optional[float] = None,
+    ) -> None:
+        super().__init__(_BayesOptim, locals(), as_config=True)
+        self.init_budget = init_budget
+        self.pca = pca
+        self.n_components = n_components
+        self.prop_doe_factor = prop_doe_factor
+
+
+PCABO = ParametrizedBayesOptim(pca=True).set_name("PCABO", register=True)
+BayesOptimBO = ParametrizedBayesOptim().set_name("BayesOptimBO", register=True)
+
+# Testing the influence of n_components on the performance of PCABO
+PCABO80 = ParametrizedBayesOptim(pca=True, n_components=0.80).set_name("PCABO80", register=True)
+
+# Testing the influence of the DoE size on the performance of PCABO
+PCABO95DoE20 = ParametrizedBayesOptim(pca=True, n_components=0.95, prop_doe_factor=0.20).set_name(
+    "PCABO95DoE20", register=True
+)
```

Comment:
If those are just for testing, either define them in the experiment directly, or in experimentalvariants.py

Reply:
Moved them to experimentalvariants.py